src/cmd/linuxkit/push_vcenter.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// Process the push arguments and execute push
+func pushVCenter(args []string) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var newVM vmConfig
+
+	flags := flag.NewFlagSet("vCenter", flag.ExitOnError)
+	invoked := filepath.Base(os.Args[0])
+
+	newVM.vCenterURL = flags.String("url", os.Getenv("VCURL"), "URL in the format of https://username:password@host/sdk")
+	newVM.dsName = flags.String("datastore", os.Getenv("VMDATASTORE"), "The Name of the DataStore to host the image")
+	newVM.networkName = flags.String("network", os.Getenv("VMNETWORK"), "The VMware vSwitch the image will use")
+	newVM.vSphereHost = flags.String("hostname", os.Getenv("VMHOST"), "The Server that will host the image")
+	newVM.path = flags.String("path", "", "Path to a specific image")
+
+	newVM.vmName = flags.String("folder", "", "A Folder on the datastore to push the image too")
+
+	flags.Usage = func() {
+		fmt.Printf("USAGE: %s push vcenter [options] [name]\n\n", invoked)
+		fmt.Printf("'name' specifies the full path of an image that will be pushed\n")
+		fmt.Printf("Options:\n\n")
+		flags.PrintDefaults()
+	}
+
+	if err := flags.Parse(args); err != nil {
+		log.Fatalln("Unable to parse args")
+	}
+
+	remArgs := flags.Args()
+	if len(remArgs) == 0 {
+		fmt.Printf("Please specify the path to the image to push\n")
+		flags.Usage()
+		os.Exit(1)
+	}
+	*newVM.path = remArgs[0]
+
+	// Ensure an iso has been passed to the vCenter push Command
+	if strings.HasSuffix(*newVM.path, ".iso") {
+		// Used as a folder to hold the images that are pushed
+		if *newVM.vmName == "" {
```

Comment:
there is no need for this to be in this condition, it can be unconditional below.